### ISSUE TITLE
[Improvement] register only the required catalogs, not all of them for spark-connector

### ISSUE THREAD

**Issue by caican00** [datastrato/gravitino#2460]
### What would you like to be improved?

All catalogs are registered during the initialization even if a spark sql does not use them.

The action may result in:
1. The list operation slows down if there are many catalogs.
2. Registering too many catalogs to memory increases memory pressure

```
com.datastrato.gravitino.spark.connector.plugin.GravitinoDriverPlugin#init

public Map<String, String> init(SparkContext sc, PluginContext pluginContext) {
    ...
    Set<String> catalogs = catalogManager.listCatalogs();
    registerGravitinoCatalogs(conf, catalogs);
    registerSqlExtensions();
    return Collections.emptyMap();
  }
```

**Comment by caican00** [comment]
cc @FANNG1

**Comment by FANNG1** [comment]
Yes, there's a trade-off, the main problem is Spark CatalogManager doesn't support load catalogs dynamically. which means we must register existing catalogs to it because we don't know which catalog which the user will use.  Fortunately there wouldn't be many catalogs normally.

**Comment by FANNG1** [comment]
#2648 , introduced a new API to load all catalogs in one call, @caican00 do you still want to work on it?
